imscjs-demo/js/index.js

```diff
@@ -1,59 +1,52 @@
-
-function initTTML() {
-  var renderDiv = document.getElementById("render-div");
-  var myVideo = document.getElementById("imscVideo");
+async function initTTML() {
+  const renderDiv = document.getElementById("render-div");
+  const myVideo = document.getElementById("imscVideo");
   //Track object generated by the first <track> child element
-  var myTrack = myVideo.textTracks[0];
+  const myTrack = myVideo.textTracks[0];
   //Src is only available from the <track> DOM element, not the obj
-  var ttmlUrl = myVideo.getElementsByTagName("track")[0].src;
+  const ttmlUrl = myVideo.getElementsByTagName("track")[0].src;
   //The track is disabled by default, 'hidden' does not display VTT Text
   //but fires the events
   myTrack.mode = "hidden";
 
-  var client = new XMLHttpRequest();
-
-  client.open('GET', ttmlUrl);
-  client.onreadystatechange = function () {
-    initTrack(client.responseText);
-  }
-  client.send();
-
+  const response = await fetch(ttmlUrl);
+  initTrack(await response.text());
 
   function initTrack(text) {
-    var imscDoc = imsc.fromXML(text);
-    var timeEvents = imscDoc.getMediaTimeEvents();
+    const imscDoc = imsc.fromXML(text);
+    const timeEvents = imscDoc.getMediaTimeEvents();
     //create cue per timed event and render isd
-    for (var i = 0; i < timeEvents.length; i++) {
+    for (let i = 0; i < timeEvents.length; i++) {
```

Comment:
Given that we modernize this example at the same time and as`timeEvents` is an `Array`. Can you double-check if we could use `for...of` here (and get rid of `i`? (Not 100% sure it is possible)

Reply:
We use `i` at https://github.com/mdn/imsc-examples/blob/943d004dfac68e5df3aecd25da461be357415171/imscjs-demo/js/index.js#L29-L31 which is why I didn't do this.